### PR TITLE
doc: update e2e README for snapshot operation

### PR DESCRIPTION
**Pull request by agarwal-mudit**
Updated e2e README with the prerequisites to run snapshot/clone e2e tests.

Signed-off-by: Mudit Agarwal <muagarwa@redhat.com>

### CODE REVIEW

File: e2e/README.md [hunk 1]
```diff
@@ -88,6 +89,31 @@ are available while running tests:
...
 | v                 | Verbose: print additional output                                              |
 
+## E2E for snapshot
+
+After the support for snapshot/clone has been added to ceph-csi,
+you need to follow these steps before running e2e.
+Please note that the snapshot operation works only if the kube version
+is greater than or equal to 1.17.0.
+
+- Delete snapshot CRD created by ceph-csi in rook
```
Comment (Madhu-1) [inline e2e/README.md:99]:
```suggestion
- Delete Alpha snapshot CRD created by ceph-csi in rook

Check if we have any `v1alpha1` CRD created in our Kubernetes cluster

```bash
[$]kubectl get crd volumesnapshotclasses.snapshot.storage.k8s.io -o yaml |grep v1alpha1
  - name: v1alpha1
  - v1alpha1
[$]kubectl get crd volumesnapshotcontents.snapshot.storage.k8s.io -o yaml |grep v1alpha1
  - name: v1alpha1
  - v1alpha1
[$]kubectl get crd volumesnapshots.snapshot.storage.k8s.io -o yaml |grep v1alpha1
  - name: v1alpha1
  - v1alpha1

If we have Alpha CRD delete it as from Kubernetes 1.17.0+ the snapshot should be `v1beta1`
```

File: e2e/README.md [hunk 2]
```diff
@@ -88,6 +89,31 @@ are available while running tests:
...
+
+    ```console
+    $./install-snapshot.sh delete-crd
+    ```
+
+- Install snapshot controller
```
Comment (Madhu-1) [inline e2e/README.md:105]:
```suggestion
- Install snapshot controller and beta  snapshot CRD
```

File: e2e/README.md [hunk 3]
```diff
@@ -88,6 +89,31 @@ are available while running tests:
 | timeout           | Panic test binary after duration d (default 0, timeout disabled)              |
 | v                 | Verbose: print additional output                                              |
 
+## E2E for snapshot
+
+After the support for snapshot/clone has been added to ceph-csi,
+you need to follow these steps before running e2e.
+Please note that the snapshot operation works only if the kube version
```
Comment (Madhu-1) [inline e2e/README.md:96]:
```suggestion
Please note that the snapshot operation works only if the Kubernetes version
```

File: e2e/README.md [hunk 4]
```diff
@@ -88,6 +89,47 @@ are available while running tests:
...
+
+      ```
+
+  - If you have Alpha CRD, delete it as from Kubernetes 1.17.0+ the snapshot should be `v1beta1`
+
+    ```console
+    $./install-snapshot.sh delete-crd
+    ```
+
+- Install snapshot controller and beta snapshot CRD
```
Comment (Madhu-1) [inline e2e/README.md:121]: sorry i should have added it as `Beta` instead of `beta` can you please correct it

**Comment by Madhu-1** [comment]
@agarwal-mudit CI failure

File: e2e/README.md [hunk 5]
```diff
@@ -88,6 +89,48 @@ are available while running tests:
...
+      [$]kubectl get crd volumesnapshotclasses.snapshot.storage.k8s.io -o yaml |grep v1alpha1
+        - name: v1alpha1
+        - v1alpha1
+      [$]kubectl get crd volumesnapshotcontents.snapshot.storage.k8s.io -o yaml |grep v1alpha1
+        - name: v1alpha1
+        - v1alpha1
+      [$]kubectl get crd volumesnapshots.snapshot.storage.k8s.io -o yaml |grep v1alpha1
+        - name: v1alpha1
+        - v1alpha1
+
```
Comment (Madhu-1) [inline e2e/README.md:112]: we can remove this extra empty line

**Comment by nixpanic** [comment]
This can be rebased when #1348 is merged. It should pass `/test commitlint` after that, in case the GitHub App does not report its status.

**Comment by nixpanic** [comment]
@mergifyio rebase

**Comment by nixpanic** [comment]
@mergifyio refresh

**Comment by nixpanic** [comment]
@mergifyio refresh

**Comment by Madhu-1** [comment]
@MergifyIO rebase

**Comment by nixpanic** [comment]
> /test commitlint

Unfortunately the commitlint CI job does not handle merge commits (done by mergify) well. A rebase is needed (or retriggering the GitHub commitlint App, somehow)

**Comment by nixpanic** [comment]
@mergifyio rebase

**Comment by nixpanic** [comment]
Did a manual rebase to drop mergify's merge commit, and in the hope that the GitHub commitlint App reports status again.

This dropped the approvals, so I'm adding mine and `[ready-to-merge]`

**Comment by nixpanic** [comment]
**ci/centos/containerized-tests** failed because:
```
import path "vbom.ml/util": reading https://vbom.ml/util?go-get=1: 503 Service Unavailable
```
which should get fixed when rerunning the job once https://vbom.ml is back up again.

**ci/centos/mini-e2e/k8s-1.17.8** failed, due to
```
cephcluster.ceph.rook.io/my-cluster created
Error from server: error when creating "https://raw.githubusercontent.com/rook/rook/v1.3.8/cluster/examples/kubernetes/ceph/toolbox.yaml": etcdserver: request timed out
Error from server (AlreadyExists): error when creating "https://raw.githubusercontent.com/rook/rook/v1.3.8/cluster/examples/kubernetes/ceph/toolbox.yaml": deployments.apps "rook-ceph-tools" already exists
...
```
which should get fixed with #1350 .

**Comment by humblec** [comment]
Looks like no luck on contenarized tests.

**Comment by humblec** [comment]
@Mergify rebase

**Comment by nixpanic** [comment]
> /skip-doc-change

This is a new CI job that I'll use for testing, ignore the **ignore/testing/skip-doc-change** job status :)

**Comment by Madhu-1** [comment]
@MergifyIO rebase

**Comment by humblec** [comment]
@Madhu-1 shall we merge ?

**Comment by nixpanic** [comment]
> /retest ci/centos/containerized-tests

The skip-doc-change piece has been merged in the CI jobs, so this **ci/centos/containerized-tests** should be quick

**Pull request merged**